codeworld-compiler/src/Compile.hs

```diff
@@ -0,0 +1,156 @@
+{-# LANGUAGE LambdaCase        #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+{-
+  Copyright 2017 The CodeWorld Authors. All rights reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-}
+
+module Compile (
+
+    compileSource
+
+)
+where
+
+import           Control.Concurrent
+import           Control.Monad
+import           Data.ByteString (ByteString)
+import qualified Data.ByteString as B
+import           System.Directory
+import           System.FilePath
+import           System.IO
+import           System.IO.Temp (withSystemTempDirectory)
+import           System.Process
+import           Text.Regex.TDFA
+
+import ErrorSanitizer
+
+compileSource :: FilePath -> FilePath -> FilePath -> [String] -> IO Bool
```

Comment:
Apart from the case of directly calling `compileSource` function, where we are sure about the argument is either going to be codeworld or haskell, we are not sure about the no. of arguments a user may pass in order to handle that part I used [String]. Is there any other better method to handle an unknown number of passed arguments to a function?